app/domain/util/open_ssl/x509/smart_cert.rb

```diff
@@ -0,0 +1,62 @@
+# A Certificate decorator that allows reading of the spiffe id and common name
+#
+require 'openssl'
+
+module Util
+  module OpenSsl
+    module X509
+      class SmartCert < SimpleDelegator
+
+        class InvalidCert < RuntimeError
```

Comment:
Are you keeping this decoupled from the big error class file to facilitate gemification?

Reply:
Definitely.  See my longer comment above.